src/internal/packager/images/pull.go

```diff
@@ -434,6 +434,44 @@ func (i *ImageConfig) PullAll() ([]ImgInfo, error) {
 	return imgInfoList, nil
 }
 
+// pullLocalImage returns a v1.Image will try to return the local tarball, this way we can isolate remote issues from non-existent local images.
+func (i *ImageConfig) pullLocalImage(src string, spinner *message.Spinner) (img v1.Image, err error) {
+	// Parse the image reference to get the image name.
+	reference, err := name.ParseReference(src)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse image reference %s: %w", src, err)
+	}
+
+	// Attempt to connect to the local docker daemon.
+	ctx := context.TODO()
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, fmt.Errorf("docker not available: %w", err)
+	}
+	cli.NegotiateAPIVersion(ctx)
+
+	// Inspect the image to get the size.
+	rawImg, _, err := cli.ImageInspectWithRaw(ctx, src)
+	if err != nil {
+		return nil, fmt.Errorf("failed to inspect image %s via docker: %w", src, err)
+	}
+
+	// Warn the user if the image is large.
+	if rawImg.Size > 750*1000*1000 {
```

Comment:
Would be nice to have a variable to avoid the magic number

Reply:
yes, I'm not fixing that right now.